stratum/hal/lib/barefoot/bfrt_node.cc

```diff
@@ -188,11 +202,12 @@ ::util::Status BfrtNode::WriteForwardingEntries(
   if (!initialized_ || !pipeline_initialized_) {
     return MAKE_ERROR(ERR_NOT_INITIALIZED) << "Not initialized!";
   }
-
+  ASSIGN_OR_RETURN(const auto& request,
+                   p4runtime_bfrt_translator_->TranslateWriteRequest(req));
   bool success = true;
   ASSIGN_OR_RETURN(auto session, bf_sde_interface_->CreateSession());
   RETURN_IF_ERROR(session->BeginBatch());
-  for (const auto& update : req.updates()) {
+  for (const auto& update : request.updates()) {
```

Comment:
I think this violates the p4rt guarantees around batching when translation fails:

> CONTINUE_ON_ERROR: This is the default behavior and the default enum value. Each operation within the batch must be attempted even if one or more encounter errors.

Should we translate one entity at a time?

Reply:
I think that makes sense. Since we should not reject the entire request if there is an invalid entry.
Do you think we can just call `TranslateEntity` at the beginning of the for-loop?

Reply:
> Do you think we can just call TranslateEntity at the beginning of the for-loop?

That should work.